### PR TITLE
chore: update effect/platform past API changes

### PR DESCRIPTION
**Pull request by riordanpawley**
~Users that are using effect/platform past a certain version will experience dependency issues if they are trying to use the latest version of uploadthing and the latest version of effect. This PR aims to resolve those issues by updating uploadthing's effect deps past the API changes.
Note that this will cause the opposite issue. People on the older effect/platform will need to update their effect deps to upgrade to the latest uploadthing.~
(edit) Today pnpm or some other part of my toolchain blessed me by no longer having the above issue

I updated effect and @effect/* deps, but they are behind again - I can update them to latest again if this PR has any interest.
I updated vite, can't remember why but I'm assuming to line up with the effect vitest types.
I turned off nuxt telemetry as it was requiring interaction during a turbo script ¯\_(ツ)_/¯

## Summary by CodeRabbit

- **New Features**
	- Introduced `FetchHttpClient` for improved HTTP handling in the upload functionality.
	- Added telemetry configuration option to minimize data collection.

- **Bug Fixes**
	- Corrected types path in shared package exports.

- **Chores**
	- Updated multiple dependencies across various packages for enhanced performance and security.
	- Updated the version of `vite` across multiple examples to improve compatibility.
	- Updated version of the `uploadthing` package to `7.1.0`.

### CODE REVIEW

File: packages/uploadthing/src/sdk/index.ts [hunk 1]
```diff
@@ -71,35 +72,44 @@ export class UTApi {
     Effect.gen(this, function* () {
       const { apiKey } = yield* UTToken;
       const baseUrl = yield* ApiUrl;
-      const httpClient = yield* HttpClient.HttpClient;
+      const httpClient = (yield* HttpClient.HttpClient).pipe(
+        HttpClient.filterStatusOk,
+      );
```
[the remark this answers is not in the document]
Comment (riordanpawley) [reply packages/uploadthing/src/sdk/index.ts:77]: this is a hallucination

File: packages/uploadthing/src/internal/upload.server.ts [hunk 2]
```diff
@@ -34,8 +32,9 @@ export const uploadWithoutProgress = (
             cause: e,
           }),
       ),
-      HttpClientResponse.json,
+      Effect.andThen((_) => _.json),
```
[the remark this answers is not in the document]
Comment (riordanpawley) [reply packages/uploadthing/src/internal/upload.server.ts:35]: hallucination

File: packages/uploadthing/src/sdk/index.ts [hunk 3]
```diff
@@ -71,44 +66,39 @@ export class UTApi {
...
-    return program.pipe(
-      Effect.provide(layer),
+    const result = await program.pipe(
       Effect.withLogSpan("utapi.#executeAsync"),
-      (e) => Effect.runPromise(e, signal ? { signal } : undefined),
+      (e) => this.runtime.runPromise(e, signal ? { signal } : undefined),
     );
+
+    await this.runtime.dispose();
+    return result;
```
[the remark this answers is not in the document]
Comment (riordanpawley) [reply packages/uploadthing/src/sdk/index.ts:101]: I agree that UTApi should provide a dispose but I don't want this PR to go into actual uploadthing user facing interface changes. This way keeps the code working the same as before while allowing dispose to be easily moved in the future.

File: packages/uploadthing/src/internal/handler.ts [hunk 4]
```diff
@@ -659,7 +651,10 @@
             onSuccess: logHttpClientResponse("Registered metadata"),
             onFailure: logHttpClientError("Failed to register metadata"),
           }),
-          HttpClientResponse.schemaBodyJsonScoped(MetadataFetchResponse),
+          Effect.flatMap(
+            HttpClientResponse.schemaBodyJson(MetadataFetchResponse),
+          ),
```
[the remark this answers is not in the document]
Comment (riordanpawley) [reply packages/uploadthing/src/internal/handler.ts:656]: just keeping functionality the same as i'm not sure why the old code had decoding after the logging

File: packages/uploadthing/src/internal/handler.ts [hunk 5]
```diff
@@ -382,13 +371,16 @@ const handleCallbackRequest = (opts: {
...
-        Effect.flatMap(HttpClient.filterStatusOk(httpClient)),
+        HttpClientRequest.bodyJson(payload),
+        Effect.flatMap(httpClient.execute),
         Effect.tapError(
           logHttpClientError("Failed to register callback result"),
         ),
-        HttpClientResponse.schemaBodyJsonScoped(CallbackResultResponse),
+        Effect.flatMap(
+          HttpClientResponse.schemaBodyJson(CallbackResultResponse),
+        ),
```
[the remark this answers is not in the document]
Comment (riordanpawley) [reply packages/uploadthing/src/internal/handler.ts:381]: just keeping functionality the same as i'm not sure why the old code had decoding after the logging

File: packages/uploadthing/src/sdk/index.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (riordanpawley) [reply packages/uploadthing/src/sdk/index.ts:101]: no, actually thinking on it further - we dont have anything to actually clean up and calling dispose is just best practices for now. lets add the dispose api if we add cleanup logic to our effects in the future

File: packages/uploadthing/src/internal/handler.ts [hunk 6]
```diff
@@ -679,3 +674,7 @@ const handleUploadAction = (opts: {
       fiber,
     };
   }).pipe(Effect.withLogSpan("handleUploadAction"));
+
+const OkHttpClient = HttpClient.HttpClient.pipe(
```
Comment (juliusmarminge) [inline packages/uploadthing/src/internal/handler.ts:678]: For the sake of consistency either remove this abstraction or use if everywhere you do `.pipe(filterOk)`

File: packages/uploadthing/src/internal/runtime.ts [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+import * as Layer from "effect/Layer";
+import * as ManagedRuntime from "effect/ManagedRuntime";
+
+import type { FetchEsque } from "@uploadthing/shared";
+
+import type { UTApiOptions } from "../types";
+import { configProvider } from "./config";
+import { withLogFormat, withMinimalLogLevel } from "./logger";
+
+export const makeRuntime = (fetch: FetchEsque, opts?: UTApiOptions) => {
```
Comment (juliusmarminge) [inline packages/uploadthing/src/internal/runtime.ts:11]: This type is misleading as we're also passing in `RouteHandlerConfig` in handler.ts.

Either an intersection here or split out a common type?

Comment (riordanpawley) [reply packages/uploadthing/src/internal/runtime.ts:11]: so a common type describing expected Env variables? alternatives would be passing a full configProvider or expecting a JSON Record

Comment (juliusmarminge) [reply packages/uploadthing/src/internal/runtime.ts]: Json record is probably the most correct, alt a ConfigProvider

Comment (juliusmarminge) [reply packages/uploadthing/src/internal/runtime.ts]: i'll do a follow-up. couldn't push to your fork

File: packages/uploadthing/src/internal/handler.ts [hunk 8]
```diff
@@ -101,13 +84,17 @@ export const makeAdapterHandler = <Args extends any[]>(
...
-    await handle.pipe(
+  return async (...args: Args) => {
+    const result = await handle.pipe(
       Effect.ap(app(...args)),
       Effect.ap(toRequest(...args)),
       Effect.withLogSpan("requestHandler"),
       managed.runPromise,
     );
+
+    await managed.dispose();
```
Comment (juliusmarminge) [inline packages/uploadthing/src/internal/handler.ts:95]: disposing here makes the runtime not available for consequent requests

```suggestion
```

**Comment by riordanpawley** [comment]
Thanks for finishing that off for me

**Pull request merged**